extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1538,6 +1540,35 @@ export class AmpA4A extends AMP.BaseElement {
    * @param {!Object<string, string|number>=} opt_extraVariables
    */
   emitLifecycleEvent(unusedEventName, opt_extraVariables) {}
+
+  /**
+   * Sets up postmessage listener for incoming messages for fluid-enabled
+   * slots. To be overriden by implementing networks.
+   * @param {Element} unusedIframe The source frame for the messages.
+   */
+  setupListenersForFluid(unusedIframe) {}
+
+  /**
+   * @param {?string} headerValue Method as given in header.
+   */
+  getNonAmpCreativeRenderingMethod(headerValue) {
+    if (headerValue) {
+      return headerValue;
+    } else if (this.experimentalNonAmpCreativeRenderMethod_) {
+      return this.experimentalNonAmpCreativeRenderMethod_;
+    }
+    return Services.platformFor(this.win).isIos() ?
+        XORIGIN_MODE.SAFEFRAME : null;
+  }
+
+  /**
+   * Returns iframe attributes necessary to render creative in cross-domain
+   * frame.
+   * @return {!JsonObject}
+   */
+  getCrossDomainFrameAttributes() {
+    return getContextMetadata(this.win, this.element, this.sentinel);
```

Comment:
I envisioned this being purely additive to the attributes already being added such that by default this returns empty.  That way you don't need to modify getContextMetadata and instead doubleclick impl could include the new values.  Let's rename this to getAdditionalContextMetadata and then we can pass it to getContextMetadata as an optional object.  That work?

Reply:
Done.